### ISSUE TITLE
question  regarding usage  of  this  library

### ISSUE THREAD

**Issue by kushal4** [Cylix/cpp_redis#89]
Hi,
My  main  confusion  is  can  this  library serve  the  purpose of  redis  as  a  whole or  it uses  some  faetures of redis.I am  mainly interested  in  using in   memory  database  form  with this  library if  possible

**Comment by Cylix** [comment]
Hi,

This library is a redis client library.
That is, this library allows you to connect to a redis server from your C++ program and to send commands to the redis server (GET, SET, HMGET, HMSET, ... for example).

If you want to build an in-memory database or a caching system or anything else using redis, you can use this library as the client, but you will need to provide the higher level logic (what should be inserted in your in-memory database, how should it be formated, ...).

In short, this library provides you communication with the redis server from your c++ software and allows you to send the commands listed on that [page](https://redis.io/commands).
Then, it is up to you to develop the highler level logic about what you want to do with redis.

Let me know if it is still unclear :)

Best!

**Comment by kushal4** [comment]
thanx :)
